signing/envelope/envelope.go

```diff
@@ -0,0 +1,56 @@
+package envelope
+
+import (
+	"encoding/json"
+
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+)
+
+/*
+Envelope captures an envelope as described by the Secure Systems Lab
+Signing Specification. See here:
+https://github.com/secure-systems-lab/signing-spec/blob/master/envelope.md
+*/
+type Envelope struct {
+	PayloadType string      `json:"payloadType"`
+	Payload     string      `json:"payload"`
+	Signatures  []Signature `json:"signatures"`
+}
+
+/*
+Signature represents a generic in-toto signature that contains the identifier
+of the key which was used to create the signature.
+The used signature scheme has to be agreed upon by the signer and verifer
+out of band.
+The signature is a base64 encoding of the raw bytes from the signature
+algorithm.
+The cert is a PEM encoded string of the signing certificate
+*/
+type Signature struct {
+	KeyID string `json:"keyid"`
+	Sig   string `json:"sig"`
+	Cert  string `json:"cert"`
+}
+
+func AddCertToEnvelope(signedAtt []byte, cert []byte) ([]byte, error) {
+	// Unmarshal into a DSSE envelope.
+	env := &dsse.Envelope{}
+	if err := json.Unmarshal(signedAtt, env); err != nil {
+		return nil, err
+	}
+
+	// Create an envelope.Envelope.
+	envWithCert := &Envelope{
+		PayloadType: env.PayloadType,
+		Payload:     env.Payload,
+		Signatures:  []Signature{},
+	}
+
+	for _, sig := range env.Signatures {
+		envWithCert.Signatures = append(envWithCert.Signatures,
+			Signature{Sig: sig.Sig, KeyID: sig.KeyID, Cert: string(cert)})
```

Comment:
no base64 encoding needed for the cert?

Reply:
It's a PEM encoded cert as noted by the comment above. The cosign cli fulcio client having a field name of `Cert` with `[]byte` type is a bit unfortunate. Should probably also be `CertPEM`.
https://github.com/sigstore/cosign/blob/97b051686a0a4fd43d78dda34c62db696340c0d1/cmd/cosign/cli/fulcio/fulcio.go#L184

Encoded by Fulcio here:
https://github.com/sigstore/fulcio/blob/23ee9bcc89178dc13781ef8c6c6fb933b5e7e01b/pkg/api/client.go#L147

Reply:
Thanks for the info. I should have guessed since it's part of a JSON string :)